### PR TITLE
refactor qwen

### PR DESCRIPTION
**Pull request by MeouSker77**
## Description

refactor qwen: use sdp_causal and new rope: 1st token perf improvement on Arc and MTL, 2nd+ token perf improvement on MTL

### CODE REVIEW

**Comment by MeouSker77** [comment]
will fix qwen logits error in next PR, for now, just use a larger error bound
